src/connector/src/source/base.rs

```diff
@@ -296,9 +298,17 @@ pub trait SplitReader: Sized + Send {
 
 for_all_sources!(impl_connector_properties);
 
+impl ConnectorProperties {
+    pub fn is_new_fs_connector(props: &BTreeMap<String, String>) -> bool {
+        props
+            .get(UPSTREAM_SOURCE_KEY)
+            .map(|s| s.eq_ignore_ascii_case(S3_V2_CONNECTOR))
```

Comment:
a little out of topic, will we support other file system in future? can we reuse the code easily?